packages/proxy/test/unit/http/util/prerequests.spec.ts

```diff
@@ -6,6 +6,11 @@ import sinon from 'sinon'
 describe('http/util/prerequests', () => {
   let preRequests: PreRequests
 
+  function expectPendingCounts (pendingRequests: number, pendingPreRequests: number) {
+    expect(preRequests.pendingRequests._length()).to.eq(pendingRequests, 'wrong number of pending requests')
+    expect(preRequests.pendingPreRequests._length()).to.eq(pendingPreRequests, 'wrong number of pending prerequests')
+  }
```

Comment:
You can run this spec and replace this fn with: 

```ts
  function expectPendingCounts (pendingRequests: number, pendingPreRequests: number) {
    expect(Object.entries(preRequests.pendingRequests).length).to.eq(pendingRequests, 'wrong number of pending requests')
    expect(Object.entries(preRequests.pendingPreRequests).length).to.eq(pendingPreRequests, 'wrong number of pending prerequests')
  }
```

to see the previous behavior, all 4 tests failed on this assertion with a leak.